### ISSUE TITLE
TransformerDocumentEmbeddings memory leak

### ISSUE THREAD

**Issue by olanag1** [flairNLP/flair#1719]
Hello, 

I'm trying to pass a list of documents and obtain document embeddings from pre-trained model "bert-base-uncased". 
To debug I measure memory consumption in for loop and I observe that on each new document the memory is leaked. 

ubuntu 18.04
flair 0.5
torch 1.5.0

> 

    document_embeddings = TransformerDocumentEmbeddings('bert-base-uncased')\n
    res = []
    i = 0
    for sent in sentences.tolist() :
        document_embeddings.embed(sent)
        res.append(pd.DataFrame(sent.embedding.detach().numpy(), columns= ['vector']) )
        print('embed doc : ', i, len(sent.tokens))
        i=i+1
        %memit  

On the 10th document the memory is off: 

> 

    embed doc :  0 5534
    peak memory: 2872.60 MiB, increment: 0.00 MiB
    embed doc :  1 5704
    peak memory: 3546.12 MiB, increment: 0.00 MiB
    embed doc :  2 13580
    peak memory: 4219.87 MiB, increment: 0.00 MiB
    embed doc :  3 17829
    peak memory: 4893.87 MiB, increment: 0.00 MiB
    embed doc :  4 10832
    peak memory: 5567.14 MiB, increment: 0.00 MiB
    embed doc :  5 15897
    peak memory: 6240.81 MiB, increment: 0.00 MiB
    embed doc :  6 3981
    peak memory: 6914.24 MiB, increment: 0.00 MiB
    embed doc :  7 11573
    peak memory: 7587.70 MiB, increment: -0.25 MiB
    embed doc :  8 6334
    peak memory: 8261.38 MiB, increment: 0.00 MiB
    embed doc :  9 5746
    peak memory: 8934.84 MiB, increment: 0.00 MiB
    embed doc :  10 7951`

I didn't find any parameter that would allow to free the memory once the computation is finished for one document. Maybe I missed something?

**Comment by buenalaune** [comment]
Hi everyone, 
first of all thanks for making this library available, it really is fun to use!
I just ran into the same problem as OP. Any advice would be greatly appreciated.
Best
Jan

**Comment by alanakbik** [comment]
Thanks for reporting this! @whoisjones can you take a look?

**Comment by buenalaune** [comment]
Hi, 
after doing some research in the issues I found this thread which solved the problem for me: https://github.com/flairNLP/flair/issues/1638
Basically, set the fine_tune flag to false when loading the model.
Thanks for the quick reply!

**Comment by olanag1** [comment]
Thanks @buenalaune, 

Setting fine_tune=False also helped in my case.  So, the memory problem is mostly linked to the finetune option.
